presto-main/src/main/java/com/facebook/presto/memory/QueryContext.java

```diff
@@ -81,6 +82,9 @@
     @GuardedBy("this")
     private long peakNodeTotalMemory;
 
+    @GuardedBy("this")
+    private Optional<Long> maxRevocableMemory;
```

Comment:
Maybe just make it `long`. No need to be optional if we set a large number for non-spilling case.